### PR TITLE
Add tests for wasm_traits file

### PR DESCRIPTION
**Pull request by findolor**
## Motivation

See issue: https://github.com/rainlanguage/rain.orderbook/issues/1799

## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [x] linked any relevant issues or PRs
- [ ] included screenshots (if this involves a front-end change)

fix https://github.com/rainlanguage/rain.orderbook/issues/1799

## Summary by CodeRabbit

- **New Features**
  - Added support for converting JavaScript BigInt values to U256, with clear error handling for invalid conversions.

- **Tests**
  - Introduced comprehensive tests to ensure correct conversion of various BigInt values, including edge cases and error scenarios.

### CODE REVIEW

File: crates/bindings/src/wasm_traits.rs [hunk 1]
```diff
@@ -1,13 +1,65 @@
+use alloy::primitives::{ruint::ParseError, U256};
 use std::str::FromStr;
+use wasm_bindgen_utils::prelude::js_sys::BigInt;
 
 // A trait for converting types to U256
```
Comment (0xgleb) [inline crates/bindings/src/wasm_traits.rs:5]: Why does this trait even exist when `TryInto` can just be implemented?

**Review by 0xgleb — APPROVED** [review]

File: crates/bindings/src/wasm_traits.rs (hunk 1 above)
Comment (findolor) [reply crates/bindings/src/wasm_traits.rs:5]: i think because `BigInt` and `U256` values are not in our local crate, try into cannot be implemented. we need to create a local trait so that it works nice with the compiler

**Review by hardyjosh — APPROVED** [review]
